### ISSUE TITLE
How to train a multi-label classifier on a big dataset ?

### ISSUE THREAD

**Issue by ErenO** [keras-team/keras#10205]
I want to train a multi-label classifier on a dataset of 300 thousand images, 
the problem is that I don't have enough memory, so I have to create a data generators to solve this problem, I have found a good link https://stanford.edu/~shervine/blog/keras-how-to-generate-data-on-the-fly.html, I have tried to transpose the code to multi-class, but I failed, 
do someone already succeed ?

**Comment by concerttttt** [comment]
Failed because of memory?
Maybe add your code here and  the crash log, let's see what happened.

**Comment by ErenO** [comment]
Hi concerttttt,

I solved my error, I knew it comes from __data_generator, I tried many possibilities until it works.

`class DataGenerator(keras.utils.Sequence):
    'Generates data for Keras'
    def __init__(self, list_IDs, labels, batch_size=32, dim=(1,150,150), n_channels=1,
                 n_classes=10, shuffle=True):
        'Initialization'
        self.dim = dim
        self.batch_size = batch_size
        self.labels = labels
        self.list_IDs = list_IDs
        self.n_channels = n_channels
        self.n_classes = n_classes
        self.shuffle = shuffle
        self.on_epoch_end()

    def __len__(self):
        'Denotes the number of batches per epoch'
        return int(np.floor(len(self.list_IDs) / self.batch_size))

    def __getitem__(self, index):
        'Generate one batch of data'
        # Generate indexes of the batch
        indexes = self.indexes[index*self.batch_size:(index+1)*self.batch_size]

        # Find list of IDs
        list_IDs_temp = [self.list_IDs[k] for k in indexes]

        # Generate data
        X, y = self.__data_generation(indexes)

        return X, y

    def on_epoch_end(self):
        'Updates indexes after each epoch'
        self.indexes = np.arange(len(self.list_IDs))
        if self.shuffle == True:
            np.random.shuffle(self.indexes)
            
    def read_and_resize(self, filepath):
        img = imread('/home/eren/dataset_scrap/' + filepath)
        res = resize(img, (150, 150), preserve_range=True, mode='reflect')
        return np.expand_dims(res, 0)
            
    def __data_generation(self, list_IDs_temp):
        'Generates data containing batch_size samples' # X : (n_samples, *dim, n_channels)
        # Initialization
        X = np.empty((self.batch_size, *self.dim, self.n_channels))
        y = np.empty((self.batch_size), dtype=int)
        
        X = [self.read_and_resize(self.list_IDs[i])
             for i in range(0, len(list_IDs_temp))]
        y = self.labels[:len(list_IDs_temp)]
        X = np.vstack(X)
        
        return X, y`

That's my previous code : 
` def __data_generation(self, list_IDs_temp):
        'Generates data containing batch_size samples' # X : (n_samples, *dim, n_channels)
        # Initialization
        X = np.empty((self.batch_size, *self.dim, self.n_channels))
        y = np.empty((self.batch_size), dtype=int)

        # Generate data
        for i, ID in enumerate(list_IDs_temp):
            X[i] = self.read_and_resize(self.list_IDs[ID])
            y[i] = self.labels[ID]
        X = np.vstack(X)
        return X, y`

**Comment by djFooFoo** [comment]
This can be closed @ErenO ?

**Comment by ErenO** [comment]
Yes
